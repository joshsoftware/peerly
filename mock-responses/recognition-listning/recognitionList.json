{
"recognitionList":[
    {
      "id":1,
      "text":
      "i give high five to avinash for writing block on react strap document",
      "given_for_user": {
            "id": 1,
            "first_name": "avinash",
            "last_name": "mane",
            "profile_image_url":"https://i.picsum.photos/id/2/200/200.jpg"
       },
      "given_by_user": {
            "id": 2,
            "first_name": "onkar",
            "last_name": "hasabe",
            "profile_image_url": "https://i.picsum.photos/id/2/200/200.jpg"
      },
      "given_at": 1234567890,
      "coreValue": {
           "id": 3,
           "text": "core value text",
           "description": "desc",
           "thumbnail": "https://i.picsum.photos/id/654/300/200.jpg"
      }
    },
     
    {
      "id":1,
      "text":"give high five to avinash writing block on react strap document",
      "given_for_user": {
            "id": 1,
            "first_name": "avinash",
            "last_name": "mane",
            "profile_image_url":"https://i.picsum.photos/id/2/200/200.jpg"
       },
      "given_by_user": {
            "id": 2,
            "first_name": "onkar",
            "last_name": "hasabe",
            "profile_image_url": "https://i.picsum.photos/id/2/200/200.jpg"
      },
     "given_at": 1234567890,
      "coreValue": {
           "id": 3,
           "text": "core value text",
           "description": "desc",
           "thumbnail_url": "https://i.picsum.photos/id/654/300/200.jpg"
      }
    }

  ]
}
