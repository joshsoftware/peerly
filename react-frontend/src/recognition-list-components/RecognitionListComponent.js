import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import RecognitionCardComponent from "recognition-list-components/RecognitionCardComponent";

const Wrapper = styled.div`
  border: 1px solid var(--grey);
  background: var(--white) 0% 0% no-repeat padding-box;
  border-radius: 36px;
  opacity: 1;
`;

const RecognitionListComponent = ({ recognitionList }) => {
  const getDateFormate = (timestamp) =>
    new Date(timestamp * 1000).toDateString();

  return (
    <Wrapper>
      {recognitionList.map((recognition, index) => (
        <RecognitionCardComponent
          key={index}
          givenByName={`${recognition.given_by_user.first_name} ${recognition.given_by_user.last_name}`}
          givenByImage={recognition.given_by_user.profile_image_url}
          givenForName={`${recognition.given_for_user.first_name} ${recognition.given_for_user.last_name}`}
          givenForImage={recognition.given_for_user.profile_image_url}
          givenAt={getDateFormate(recognition.given_at)}
          text={recognition.text}
          coreValue={recognition.coreValue.text}
          coreValueImage={recognition.coreValue.thumbnail_url}
        />
      ))}
    </Wrapper>
  );
};

RecognitionListComponent.propTypes = {
  recognitionList: PropTypes.array.isRequired,
};

export default React.memo(RecognitionListComponent);
