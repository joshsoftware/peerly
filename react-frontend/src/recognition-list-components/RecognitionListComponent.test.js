import React from "react";
import { render } from "@testing-library/react";

import RecognitionListComponent from "recognition-list-components/RecognitionListComponent";
import recognitionList from "../../../mock-responses/recognition-listning/recognitionList.json";

describe("Recognition list Component test", () => {
  test("should equal snapshot", () => {
    const { asFragment } = render(
      <RecognitionListComponent
        recognitionList={recognitionList.recognitionList}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
