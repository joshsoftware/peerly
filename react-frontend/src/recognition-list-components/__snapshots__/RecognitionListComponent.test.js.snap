// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recognition list Component test should equal snapshot 1`] = `
<DocumentFragment>
  .c3 {
  width: 12vh;
}

.c2 {
  color: var(--black);
  background-color: var(--white);
  font-weight: bold;
  border: none;
}

.c5 {
  border: 1px solid var(--grey);
  border-radius: 11px;
  opacity: 1;
  margin-left: 5%;
  margin-right: 5%;
}

.c6 {
  width: auto;
  margin-left: 10%;
}

.c7 {
  width: 7vh;
  margin-right: 5%;
}

.c4 {
  border-radius: 20px;
  overflow: hidden;
}

.c8 {
  position: absolute;
  bottom: 50px;
}

.c1 {
  border-radius: 20px;
  margin: 20px 20px;
  box-shadow: 0px 5px 20px var(--box-shadow-color);
  opacity: 1;
  background: var(--white) 0% 0% no-repeat padding-box;
}

.c0 {
  border: 1px solid var(--grey);
  background: var(--white) 0% 0% no-repeat padding-box;
  border-radius: 36px;
  opacity: 1;
}

<div
    class="c0"
  >
    <div
      class="c1 card"
    >
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <div
            class="col-md-7 col-sm-12"
          >
            <div>
              <div
                class="d-flex justify-content-end"
              >
                <button
                  class="c2 btn btn-primary"
                  type="button"
                >
                  ...
                </button>
              </div>
              <div
                class="d-flex flex-row"
              >
                <div
                  class="c3"
                >
                  <img
                    alt="Profile"
                    class="img-fluid rounded-circle"
                    src="https://i.picsum.photos/id/2/200/200.jpg"
                  />
                </div>
                <div
                  class="text-start ml-2"
                >
                  <label
                    class="font-weight-bold  form-label"
                  >
                    avinash mane
                  </label>
                  <div>
                    <label
                      class="form-label"
                    >
                       got a high five for
                    </label>
                    <label
                      class="ml-1 font-weight-bold form-label"
                    >
                      core value text
                    </label>
                  </div>
                  <label
                    class="text-muted form-label"
                  >
                    Sat Feb 14 2009
                  </label>
                </div>
              </div>
            </div>
            <img
              alt="Core value"
              class="c4 d-sm-block d-xs-block d-md-none mt-2 img-fluid"
              src="https://i.picsum.photos/id/654/300/200.jpg"
            />
            <div>
              <fieldset
                class="c5"
              >
                <legend
                  class="c6"
                >
                  “
                </legend>
                <label
                  class="px-2 pb-2 pt-0 form-label"
                >
                  i give high five to avinash for writing block on react strap document
                </label>
              </fieldset>
              <div
                class="d-flex justify-content-end w-100"
              >
                <label
                  class="font-weight-bold text-dark form-label"
                >
                  onkar hasabe
                </label>
                <img
                  alt="Profile"
                  class="c7 img-fluid rounded-circle"
                  src="https://i.picsum.photos/id/2/200/200.jpg"
                />
              </div>
            </div>
          </div>
          <div
            class="d-none d-md-block text-center col"
          >
            <div>
              <img
                alt="Core value"
                class="c4 mt-2 img-fluid"
                src="https://i.picsum.photos/id/654/300/200.jpg"
              />
            </div>
            <svg
              class="mt-4"
              color="red"
              fill="currentColor"
              height="50px"
              stroke="currentColor"
              stroke-width="0"
              style="color: red;"
              viewBox="0 0 16 16"
              width="50px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8 15A7 7 0 108 1a7 7 0 000 14zm0 1A8 8 0 108 0a8 8 0 000 16z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
        <img
          alt="High five recognition"
          class="c8 img-fluid"
          src="/assets/images//high-five.png"
        />
        <div
          class="bg-white card-footer"
        >
          <span
            class="font-weight-bold text-muted"
          >
            +
          </span>
          <span
            class="font-weight-bold text-dark"
          >
            1
          </span>
        </div>
      </div>
    </div>
    <div
      class="c1 card"
    >
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <div
            class="col-md-7 col-sm-12"
          >
            <div>
              <div
                class="d-flex justify-content-end"
              >
                <button
                  class="c2 btn btn-primary"
                  type="button"
                >
                  ...
                </button>
              </div>
              <div
                class="d-flex flex-row"
              >
                <div
                  class="c3"
                >
                  <img
                    alt="Profile"
                    class="img-fluid rounded-circle"
                    src="https://i.picsum.photos/id/2/200/200.jpg"
                  />
                </div>
                <div
                  class="text-start ml-2"
                >
                  <label
                    class="font-weight-bold  form-label"
                  >
                    avinash mane
                  </label>
                  <div>
                    <label
                      class="form-label"
                    >
                       got a high five for
                    </label>
                    <label
                      class="ml-1 font-weight-bold form-label"
                    >
                      core value text
                    </label>
                  </div>
                  <label
                    class="text-muted form-label"
                  >
                    Sat Feb 14 2009
                  </label>
                </div>
              </div>
            </div>
            <img
              alt="Core value"
              class="c4 d-sm-block d-xs-block d-md-none mt-2 img-fluid"
              src="https://i.picsum.photos/id/654/300/200.jpg"
            />
            <div>
              <fieldset
                class="c5"
              >
                <legend
                  class="c6"
                >
                  “
                </legend>
                <label
                  class="px-2 pb-2 pt-0 form-label"
                >
                  give high five to avinash writing block on react strap document
                </label>
              </fieldset>
              <div
                class="d-flex justify-content-end w-100"
              >
                <label
                  class="font-weight-bold text-dark form-label"
                >
                  onkar hasabe
                </label>
                <img
                  alt="Profile"
                  class="c7 img-fluid rounded-circle"
                  src="https://i.picsum.photos/id/2/200/200.jpg"
                />
              </div>
            </div>
          </div>
          <div
            class="d-none d-md-block text-center col"
          >
            <div>
              <img
                alt="Core value"
                class="c4 mt-2 img-fluid"
                src="https://i.picsum.photos/id/654/300/200.jpg"
              />
            </div>
            <svg
              class="mt-4"
              color="red"
              fill="currentColor"
              height="50px"
              stroke="currentColor"
              stroke-width="0"
              style="color: red;"
              viewBox="0 0 16 16"
              width="50px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8 15A7 7 0 108 1a7 7 0 000 14zm0 1A8 8 0 108 0a8 8 0 000 16z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
        <img
          alt="High five recognition"
          class="c8 img-fluid"
          src="/assets/images//high-five.png"
        />
        <div
          class="bg-white card-footer"
        >
          <span
            class="font-weight-bold text-muted"
          >
            +
          </span>
          <span
            class="font-weight-bold text-dark"
          >
            1
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
